mephisto/client/review_app/server/api/views/stats_view.py

```diff
@@ -0,0 +1,191 @@
+    worker_id: Optional[str] = None,
+    task_id: Optional[str] = None,
+    statuses: Optional[List[str]] = None,
+    since: Optional[str] = None,
+    limit: Optional[int] = None,
+) -> List[StringIDRow]:
+    params = []
+
+    worker_query = "worker_id = ?" if worker_id else ""
+    if worker_id:
+        params.append(nonesafe_int(worker_id))
+
+    task_query = "task_id = ?" if task_id else ""
+    if task_id:
+        params.append(nonesafe_int(task_id))
+
+    statuses_string = ','.join([f"'{s}'" for s in statuses])
+    status_query = f"status IN ({statuses_string})" if statuses else ""
+
+    since_query = "creation_date >= ?" if since else ""
+    if since:
+        params.append(since)
+
+    joined_queries = ' AND '.join(list(filter(bool, [
+        worker_query, task_query, status_query, since_query,
+    ])))
+
+    where_query = f"WHERE {joined_queries}" if joined_queries else ""
+
+    limit_query = "LIMIT ?" if limit else ""
+    if limit:
+        params.append(nonesafe_int(limit))
+
+    with db.table_access_condition:
+        conn = db._get_connection()
+        conn.set_trace_callback(print)
+        c = conn.cursor()
+        c.execute(
+            f"""
+            SELECT * FROM units
+            {where_query}
+            ORDER BY creation_date ASC {limit_query};
+            """,
+            params,
+        )
+
+        results = c.fetchall()
+        return results
+
+
+class StatsView(MethodView):
+    def get(self) -> dict:
+        """ Get stats of recent approvals for the worker or task """
+
+        worker_id = request.args.get("worker_id")
+        task_id = request.args.get("task_id")
+        limit = request.args.get("limit")
+        since = request.args.get("since")
+
+        # Validate `since` and change format
+        if since:
+            try:
+                since = parse(since).strftime("%Y-%m-%d %H:%M:%S")
+            except ParserError:
+                raise BadRequest("Wrong date format.")
+
+        approved_unit_reviews = _find_unit_reviews(
+            db=app.db,
+            worker_id=worker_id,
+            task_id=task_id,
+            status=AssignmentState.ACCEPTED,
+            since=since,
+            limit=limit,
+        )
+        rejected_unit_reviews = _find_unit_reviews(
+            db=app.db,
+            worker_id=worker_id,
+            task_id=task_id,
+            status=AssignmentState.REJECTED,
+            since=since,
+            limit=limit,
+        )
+        soft_rejected_unit_reviews = _find_unit_reviews(
+            db=app.db,
+            worker_id=worker_id,
+            task_id=task_id,
+            status=AssignmentState.SOFT_REJECTED,
+            since=since,
+            limit=limit,
+        )
+        all_unit_reviews = _find_units_for_worker(
+            db=app.db,
+            worker_id=worker_id,
+            task_id=task_id,
+            statuses=AssignmentState.completed(),
+            since=since,
+            limit=limit,
+        )
```

Comment:
Would it be better to get all the units at once, then filter by status? Unsure if this is going to incur any additional search cost or not depending on the DB.